features/profile/view/NewHostRequest.tsx

```diff
@@ -115,9 +129,13 @@ export default function NewHostRequest({
     );
   });
 
-  const watchFromDate = watch("fromDate", dayjs());
+  const watchFromDate = watch("fromDate", undefined);
   useEffect(() => {
-    if (isSameOrFutureDate(watchFromDate, getValues("toDate"))) {
+    if (
+      watchFromDate &&
+      getValues("toDate") &&
+      isSameOrFutureDate(watchFromDate, getValues("toDate"))
+    ) {
       setValue("toDate", watchFromDate.add(1, "day"));
     }
   });
```

Comment:
Appreciate this is not code you've written (and it's been done a while ago!), but this `useEffect` feels like the wrong way to sync the `toDate` (departure date) field whenever the `fromDate` (arrival date) changes, as it's tied to a user interaction (when someone clicks and populates the date picker) and so the sync should be moved into the event handler imo. There's a `onPostChange` on the Datepicker component that would be suitable for the job!

(But also appreciate you may not want to change too much here for a small fix, so I let you decide if you want to make the change or not 🙂 )

Reply:
I prefer to not 🙃